src/NgacApi/NgacApi/attribute.py

```diff
@@ -1,94 +1,101 @@
-
-
-class ObjectAttribute(Attribute):
-    """
-    ObjectAttribute abstraction
-    ---
-
-    Used to described attributes for `Resource`s.
-    """
-
-    def __init__(self, object_attribute: str):
-        # super.__init__(obj_attr)
-        self.object_attribute = object_attribute
-
-    def __str__(self) -> str:
-        return self.object_attribute
-
-
-class UserAttribute(Attribute):
-    """
-    UserAttribute abstraction
-    ---
-
-    Used to describe attributes for `User`s.
-    """
-
-    def __init__(self, user_attr: str):
-        self.user_attr = user_attr
-
-    def __str__(self) -> str:
-        return self.user_attr
-
-
-def test_ngac_attribute():
-    """
-    Test attribute class.
-    """
-    attr = ObjectAttribute("SomeAttribute")
-    assert attr.get_attribute() == "SomeAttribute"
-
-    attr2 = UserAttribute("SomeAttribute")
-    assert attr2.get_attribute() == "SomeAttribute"
-
-    assert attr == attr2
-
-
-def test_attr_repr():
-    """
-    Test attribute class.
-    """
-    attr = ObjectAttribute("SomeAttribute")
-    assert attr.__repr__() == "SomeAttribute"
-
-    attr2 = UserAttribute("SomeAttribute")
-    assert attr2.__repr__() == "SomeAttribute"
-
-    assert attr == attr2
-    assert attr.__hash__() == attr2.__hash__()
-    assert attr.__str__() == attr2.__str__()
+"""
+NGAC attribute
+---
+
+This file describes a simple attribute abstraction as well as 2 wrapper classes that are used for the `User` and `Resource` classes.
+
+The `Attribute` class it self should never be used.
+"""
+
+
+from .ngac_object import NgacObject
+from .policy_element import PolicyElement
+
+
+class Attribute(NgacObject):
+    """
+    NGAC attribute abstraction class
+    ---
+
+    This class is only for typing.
+    """
+
+    def __str__(self) -> str:
+        return ""
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+    def get_attribute(self) -> str:
+        return self.__str__()
+
+    def __eq__(self, __o: "Attribute") -> bool:
+        return self.__str__() == __o.__str__()  # They are just string wrappers
+
+    def __hash__(self) -> int:
+        return hash(self.__str__())
+
+
+class ObjectAttribute(Attribute, PolicyElement):
+    """
+    ObjectAttribute abstraction
+    ---
+
+    Used to described attributes for `Resource`s.
+    """
+
+    def __init__(self, object_attribute: str):
+        # super.__init__(obj_attr)
+        self.object_attribute = object_attribute
+
+    def __str__(self) -> str:
+        return self.object_attribute
+
+    def pol_el_repr(self) -> str:
+        return f"object_attribute({self.object_attribute})"
+
+
+class UserAttribute(Attribute, PolicyElement):
+    """
+    UserAttribute abstraction
+    ---
+
+    Used to describe attributes for `User`s.
+    """
+
+    def __init__(self, user_attr: str):
+        self.user_attr = user_attr
+
+    def __str__(self) -> str:
+        return self.user_attr
+
+    def pol_el_repr(self) -> str:

```

Comment:
It is an abbreviation of `policy_element_representation` which I thought would be too long. It might be fruitful to add a doc comment explaining it though. However, it is not meant to be used outside of the NGAC library so I am not sure if it's worth cluttering the source code, what do you think?

Reply:
The function is used to "unpack" the elements that are policy elements to the representation that the policy expects, the user object for example, is a composite element of a bunch of attribute assigns and the user id, to create a new user the user has to be "unpacked" to one `user(uid)` and a bunch of `assign(uid,uaid)`

Reply:
Okey, then the name makes sense. Then I do not think that a docstring is needed. This is probably something that I should have been able to figure out.

Reply:
No worries, It's a massive PR so it's quite hard to in a reasonable amount of time get a perfect understanding of the code